apps/evm/ui/pool/RemoveSection/RemoveSectionLegacy.tsx

```diff
@@ -160,87 +163,97 @@ export const RemoveSectionLegacy: FC<RemoveSectionLegacyProps> = withCheckerRoot
-            address,
-            deadline.toHexString(),
-          ]
-        } else {
-          methodNames = ['removeLiquidity']
-          args = [
-            pool.token0.wrapped.address,
-            pool.token1.wrapped.address,
-            balance[FundSource.WALLET].multiply(percentToRemove).quotient.toString(),
-            minAmount0.quotient.toString(),
-            minAmount1.quotient.toString(),
-            address,
-            deadline.toHexString(),
-          ]
-        }
+            return {
+              functionNames: ['removeLiquidityETH', 'removeLiquidityETHSupportingFeeOnTransferTokens'],
+              args: [
+                token1IsNative ? (pool.token0.wrapped.address as Address) : (pool.token1.wrapped.address as Address),
+                balance[FundSource.WALLET].multiply(percentToRemove).quotient,
+                token1IsNative ? minAmount0.quotient : minAmount1.quotient,
+                token1IsNative ? minAmount1.quotient : minAmount0.quotient,
+                address,
+                deadline,
+              ],
+            } as const
+          }
+
+          return {
+            functionNames: ['removeLiquidity'],
+            args: [
+              pool.token0.wrapped.address as Address,
+              pool.token1.wrapped.address as Address,
+              balance[FundSource.WALLET].multiply(percentToRemove).quotient,
+              minAmount0.quotient,
+              minAmount1.quotient,
+              address,
+              deadline,
+            ],
+          } as const
+        })()
 
         const safeGasEstimates = await Promise.all(
-          methodNames.map((methodName) =>
-            contract.estimateGas[methodName](...args)
+          config.functionNames.map((methodName) =>
+            contract.estimateGas[methodName](...(config.args as any))
               .then(calculateGasMargin)
               .catch(() => undefined)
           )
         )
 
-        const indexOfSuccessfulEstimation = safeGasEstimates.findIndex((safeGasEstimate) =>
-          BigNumber.isBigNumber(safeGasEstimate)
+        const indexOfSuccessfulEstimation = safeGasEstimates.findIndex(
+          (safeGasEstimate) => typeof safeGasEstimate === 'bigint'
         )
 
         if (indexOfSuccessfulEstimation !== -1) {
-          const methodName = methodNames[indexOfSuccessfulEstimation]
+          const methodName = config.functionNames[indexOfSuccessfulEstimation]
           const safeGasEstimate = safeGasEstimates[indexOfSuccessfulEstimation]
 
-          setRequest({
-            from: address,
+          return {
+            account: address,
             to: contract.address,
-            data: contract.interface.encodeFunctionData(methodName, args),
-            gasLimit: safeGasEstimate,
-          })
+            data: encodeFunctionData({ abi: contract.abi, functionName: methodName, args: config.args as any }),
```

Comment:
as any?

Reply:
Same as in the other case, a lot of TS magic required to fix, been down the rabbit hole